File: 387_First_Unique_Character_in_a_String.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        mp = {}
        
        for char in s:
            if char not in mp:
                mp[char] = 0
            else:
                mp[char] += 1
        
        for i in range(len(s)):
            if mp[s[i]] == 0:
                return i
        
        return -1
        

File: swift/387_first_unique_character_in_a_string.swift
class Solution {
  func firstUniqChar(_ s: String) -> Int {
    let charCounter: [Character: Int] = createCounter(of: s)

    for (curIdx, char) in s.enumerated() {
      guard let count = charCounter[char], count != 1 else {
        return curIdx
      }
    }

    return -1
  }

  private func createCounter(of str: String) -> [Character: Int] {
    var counter: [Character: Int] = [:]

    for char in str {
      counter[char, default: 0] += 1
    }

    return counter
  }
}
